SofaKernel/framework/sofa/helper/vector.h

```diff
@@ -234,170 +316,132 @@ inline std::istream& vector<int >::read( std::istream& in )
                     t2 = t;
                 }
             }
+
+            /// Go in backward order.
             if (tinc < 0)
                 for (t=t1; t>=t2; t+=tinc)
                     this->push_back(t);
+            /// Go in Forward order
             else
                 for (t=t1; t<=t2; t+=tinc)
                     this->push_back(t);
         }
     }
     if( in.rdstate() & std::ios_base::eofbit ) { in.clear(); }
-    return in;
-}
-
-/// Output stream
-/// Specialization for writing vectors of unsigned char
-template<>
-inline std::ostream& vector<unsigned char >::write(std::ostream& os) const
-{
-    if( this->size()>0 )
-    {
-        for( size_type i=0; i<this->size()-1; ++i )
-            os<<(int)(*this)[i]<<" ";
-        os<<(int)(*this)[this->size()-1];
-    }
-    return os;
-}
-
-/// Inpu stream
-/// Specialization for writing vectors of unsigned char
-template<>
-inline std::istream&  vector<unsigned char >::read(std::istream& in)
-{
-    int t;
-    this->clear();
-    while(in>>t)
+    if(numErrors!=0)
     {
-        this->push_back((unsigned char)t);
+        msg_warning("vector<int>") << "Unable to parse vector values:" << msgendl
+                                   << msg.str() ;
     }
-    if( in.rdstate() & std::ios_base::eofbit ) { in.clear(); }
     return in;
 }
 
+
 /// Input stream
 /// Specialization for reading vectors of int and unsigned int using "A-B" notation for all integers between A and B
-template<>
-inline std::istream& vector<unsigned int >::read( std::istream& in )
+template<> inline
+std::istream& vector<unsigned int>::read( std::istream& in )
 {
-    unsigned int t;
+    std::stringstream errmsg ;
+    unsigned int errcnt = 0 ;
+    unsigned int t = 0 ;
+
     this->clear();
     std::string s;
     while(in>>s)
     {
         std::string::size_type hyphen = s.find_first_of('-',1);
         if (hyphen == std::string::npos)
         {
-            t = atoi(s.c_str());
+            t = getUnsignedInteger(s, errmsg, errcnt) ;
```

Comment:
looking for '-' has already been done, and you do it again in getUnsignedInteger